### PR TITLE
Less aggressive autorefactoring with proper code style set up from the start

### PR DESCRIPTION
**Pull request by hyphz**
This should involve less churn.

### CODE REVIEW

File: code/src/java/pcgen/cdom/choiceset/AbilityFromClassChoiceSet.java [hunk 1]
```diff
@@ -135,28 +135,22 @@ public int hashCode()
...
+            {
+                set.add(new CNAbilitySelection(
+                    CNAbilityFactory.getCNAbility(AbilityCategory.FEAT, Nature.VIRTUAL, aFeat)));
+            }
+            for (int lvl = 0; lvl < pc.getLevel(aClass); lvl++)
 			{
 				PCClassLevel pcl = pc.getActiveClassLevel(aClass, lvl);
 				//TODO This is a bug -> it was not properly gathering before
 				abilityList = Collections.emptyList();
-				if (abilityList != null)
```
Comment (hyphz) [inline code/src/java/pcgen/cdom/choiceset/AbilityFromClassChoiceSet.java]: Builtin Collections.emptyList() will not return null.

File: code/src/java/pcgen/cdom/content/LevelCommandFactory.java [hunk 2]
```diff
@@ -148,11 +148,7 @@ public int compareTo(LevelCommandFactory other)
 			{
 				return 0;
 			}
-			if (levels == other.levels)
```
Comment (hyphz) [inline code/src/java/pcgen/cdom/content/LevelCommandFactory.java]: If these are equal, compareTo will return 0 anyway (doesn't seem to imply there should be a special case for reference equality)

File: code/src/java/pcgen/cdom/enumeration/GroupingState.java [hunk 3]
```diff
@@ -81,7 +81,7 @@ public GroupingState add(GroupingState state)
 					"Attempt to add '" + state + "' grouping state to " + "'ALLOWS_NONE' resulted in 'INVALID'.");
 				return INVALID;
 			}
-			return state == EMPTY ? ALLOWS_NONE : INVALID;
```
Comment (hyphz) [inline code/src/java/pcgen/cdom/enumeration/GroupingState.java]: If state!=EMPTY we already returned from the IF above.

File: code/src/java/pcgen/cdom/facet/RemoteModifierFacet.java [hunk 4]
```diff
@@ -122,15 +122,10 @@ else if (sourceFormatManager.isPresent())
 			returnValue = new DefinedWrappingModifier<>(modifier, "source",
 				source, sourceFormatManager.get());
 		}
-		else if (sourceFormatManager.isPresent())
```
Comment (hyphz) [inline code/src/java/pcgen/cdom/facet/RemoteModifierFacet.java]: Doubled IF condition.

File: code/src/java/pcgen/cdom/formula/ListChannelAdapter.java [hunk 5]
```diff
@@ -119,7 +119,7 @@ public void referenceChanged(ReferenceEvent<T[]> e)
 		List<T> oldList = Arrays.asList(oldArray);
 		T[] newArray = e.getNewReference();
 		List<T> newList = Arrays.asList(newArray);
-		for (int i = oldArray.length; i >= 0 ; i--)
+		for (int i = oldArray.length;true;i--)
```
Comment (hyphz) [inline code/src/java/pcgen/cdom/formula/ListChannelAdapter.java]: Autoref error. Reverted in later commit

File: code/src/java/pcgen/cdom/enumeration/ListKey.java [hunk 6]
```diff
@@ -329,27 +329,26 @@ private static void buildMap()
...
-					throw new UnreachableError(e);
-				}
-			}
-		}
+            if (java.lang.reflect.Modifier.isStatic(mod) && java.lang.reflect.Modifier.isFinal(mod)
+                    && java.lang.reflect.Modifier.isPublic(mod))
+            {
+                try
+                {
+                    Object obj = field.get(null);
```
Comment (mertonmonk) [inline code/src/java/pcgen/cdom/enumeration/ListKey.java]: what is the point of field.get(null)?

**Comment by LegacyKing** [comment]
@mertonmonk are you able to merge?

**Comment by LegacyKing** [comment]
@mertonmonk use Squash and merge option if able...

**Comment by mertonmonk** [comment]
Was this PR merged?

**Comment by mertonmonk** [comment]
I don't see any options.  Rebase and merge is disabled for this PR.  There's only a Close and comment button that is enabled.

**Comment by hyphz** [comment]
Was it due to the "unresolved" explanations I added?

**Comment by LegacyKing** [comment]
Practice... Use the Drop Down to select the other available option. I've now merged as you've indicated this is approved.

**Pull request merged**